### ISSUE TITLE
Out of 100 titles, only 37 are being identified.

### ISSUE THREAD

**Issue by mizifih** [a1ex4/ownfoil#132]
### Discussed in https://github.com/a1ex4/ownfoil/discussions/131

<div type='discussions-op-text'>

<sup>Originally posted by **mizifih** October 16, 2024</sup>
I'm using Nsx Library Manager to manage my library identifying Games, Updates and DLCs. The numbers I'm using are based on Nsx Library Manager identification and organization, there isn't a single unidentified file in my library.

But, Ownfoil isn't recognizing all of them.

It was able to identify:
- 37 games out of 100
- 208 DLCs out of 235
- Can't list Updates in Ownfoil

What am I'm missing? Can it be something on my naming schemes causing this difference in discoverability? All files are named following the same patter, also, I'm adding Title ID, Patch ID and Version to the file name when possible.

I have NSP, NSZ, XCI and XCZ files. These is are the naming schemes I'm using:

- **GAME:** `[BASE]{TitleName}.TID-[{TitleId}].PID-[{PatchId}].{Extension}`
  - [BASE]Cat Quest III.TID-[010088501B8F2000].PID-[010088501B8F2800].nsp
- **UPDATE:** `[UPD]{TitleName}.TID-[{TitleId}].VER-[{Version}].{Extension}`
  - [UPD]Bayonetta 3.TID-[01004A4010FEA800].VER-[65536].nsp
- **DLC:** `[DLC]{AppName}.{TitleName}.TID-[{TitleId}].{Extension}`
  - [DLC]Super Smash Bros. Ultimate.【Costume】Erdrick's Armor and Helmet.TID-[01006A800016F025].nsp

### Here's an exemple of something not being listed in Ownfoil:
**FILES**
```
[BASE]Cat Quest III.TID-[010088501B8F2000].PID-[010088501B8F2800].nsp
[UPD]Cat Quest III.TID-[010088501B8F2800].VER-[393216].nsp
[UPD]Cat Quest III.TID-[010088501B8F2800].VER-[131072].nsp
[UPD]Cat Quest III.TID-[010088501B8F2800].VER-[327680].nsp
[UPD]Cat Quest III.TID-[010088501B8F2800].VER-[196608].nsp
```
![image](https://github.com/user-attachments/assets/0ca077b6-616a-4f3b-824e-aa02280cebb2)

**LOG (search for "_cat_")**
```
[2024-10-16 14:02:58.101] WARNING (library) File contains None value, it will be skipped: {'id': 511, 'filepath': '/games/Nintendo - Switch/[UPD]Cat Quest III.TID-[010088501B8F2800].VER-[327680].nsp', 'library': '/games/Nintendo - Switch', 'folder': '', 'filename': '[UPD]Cat Quest III.TID-[010088501B8F2800].VER-[327680].nsp', 'title_id': '010088501B8F2000', 'app_id': '010088501B8F2800', 'type': 'UPDATE', 'version': None, 'extension': 'nsp', 'size': 304101144, 'identification': 'filename'}

[2024-10-16 14:02:58.163] WARNING (library) File contains None value, it will be skipped: {'id': 514, 'filepath': '/games/Nintendo - Switch/[UPD]Cat Quest III.TID-[010088501B8F2800].VER-[196608].nsp', 'library': '/games/Nintendo - Switch', 'folder': '', 'filename': '[UPD]Cat Quest III.TID-[010088501B8F2800].VER-[196608].nsp', 'title_id': '010088501B8F2000', 'app_id': '010088501B8F2800', 'type': 'UPDATE', 'version': None, 'extension': 'nsp', 'size': 303154968, 'identification': 'filename'}
```

**Comment by ParadoxInfinite** [comment]
@mizifih It could be the file naming structure, from the README, it requires using `[APP_ID][vVERSION]` format, notice the `v` before the version number. Try adding a `v` before your numbers and check if that detects the games.
Ex:
Rename
`[BASE]Cat Quest III.TID-[010088501B8F2000].PID-[010088501B8F2800].nsp`
to
`[BASE]Cat Quest III.TID-[010088501B8F2000].PID-[v010088501B8F2800].nsp`

**Comment by a1ex4** [comment]
Closing as resolved in the discussion #131.
